resource_dispatcher/src/server.py

```diff
@@ -43,10 +43,14 @@ def sync(self, parent, children):
                 )
                 return {"status": {}, "children": []}
 
-            desired_secret_count = len(glob.glob(f"{folder}/*.yaml"))
+            desired_secrets_count = len(glob.glob(f"{folder}/secrets/*.yaml"))
+            desired_svc_accounts_count = len(glob.glob(f"{folder}/service-accounts/*.yaml"))
```

Comment:
This check compares the desired_X_count to the number of **files** for that resource type.  Is it defined somewhere that each .yaml file can contain only one resource?  That feels like an unexpected constraint (kubernetes typically accepts many resources in the same yaml).  

Do we need this constraint?  Couldn't we just count the resources of each type a few lines from now once the manifests are loaded?

Reply:
If we do have this constraint, are you sure the resource dispatcher charm always follows this?  I am reading through and I'm not sure

Reply:
So I added the first idea of iterating and counting kinds after loading the resource files. But there is no way there will be a file that contains two yamls as the resource dispatcher receives in relation list of yamls which are generated from python dicts ... then resource dispatcher charm itters through that list and writes each object one by one. The only place were a problem may occur is the mlflow or any charm joining the relation as the command `yaml.safe_load()` cant read yaml with multiple manifests. but here its ok

Reply:
Lets accept this for now, and I'll poke at the code more.  

I don't know if this breaks anything here, but in general it feels like a better practice to count the number of objects rather than files.  You are probably right that the charm using this will only do one resource per file, but that's still an undocumented convention that feels odd.  It is typical for k8s manifests to have multiple resources, so it feels better to just stay inside that convention.

If it was hard to compute the object count here, I'd be more sympathetic.  But in this case it is really easy.  If we move the `desired_secrets_count = ___` down below the `desired_resources` computation, then you can just count the objects themselves rather than use the files as a surrogate.  This also means you don't need to trust the user puts the right resource in the right subfolder (or that this main.py need to know anything about the subfolder structure at all).  

It isn't that the current way is broken, but in the same amount of code you could make things more flexible by refactoring a little.

Reply:
I added it :)